### PR TITLE
CVE Batch Job for updating the cve_source property as CRA

### PR DESCRIPTION
**Pull request by animuk**
The purpose of this Batch Job which is going to be invoked from Gemini is to sync all the CVE nodes and update the cve_source input property as the value given and the ecosystem given as input.

### CODE REVIEW

File: src/utils.py [hunk 1]
```diff
@@ -262,6 +262,52 @@ def sync_all_non_cve_version(input):
...
+    logger.info("Sync called for CVEs to update the latest cve sources.")
+    resp = {
+        "message": "Latest cve source rectified for the CVEs",
+        "status": "Success"
+    }
+    ecosystems = input.get("ecosystems", [{0: 0}])
+    cve_sources = input['cve_sources']
+    for eco in ecosystems:
+        # Fetch all the cve ids for the ecosystem
+        logger.info(f'Updating cve source as {cve_sources} for {eco}')
```
Comment (yzainee) [inline src/utils.py]: There seems to be a typo here

File: src/utils.py [hunk 2]
```diff
@@ -262,6 +262,52 @@ def sync_all_non_cve_version(input):
     return resp
 
 
+def sync_all_cve_source(input):
```
Comment (yzainee) [inline src/utils.py]: i dont see any reason to repeat this functionality. This is already present. Just modify it to call the rectify_cve_source function on the basis of input sent. Please remove this function.
refer to function : sync_all_non_cve_version

File: src/utils.py [hunk 3]
```diff
@@ -262,6 +262,52 @@ def sync_all_non_cve_version(input):
...
+def rectify_cve_source(input, cve_sources):
+    """Rectify the latest version of the EPVs."""
+    query_str = "g.V().has('cve_id', '{arg0}')" \
+                ".property('cve_sources', '{arg1}').iterate();"
+    args = []
+    resp = {
+        "message": "cve sources updated for the CVEs",
+        "status": "Success"
+    }
+    for cve in input:
```
Comment (yzainee) [inline src/utils.py]: Add a new if condition in the batch query executor. Doesnt feel right to pass 2 blank values

File: test/test_utils.py [hunk 4]
```diff
@@ -287,6 +288,36 @@ def test_sync_all_non_cve_version(mock1, mock2, mock3, mock4):
...
+    """Test sync_all_non_cve_version function."""
+    input = ['CVE-2018-3717']
+    mock1.return_value = None
+
+    res = rectify_cve_source(input, 'CRA')
+    assert res["message"] == "cve sources updated for the CVEs"
+    assert res["status"] == "Success"
+
+
 if __name__ == '__main__':
```
Comment (yzainee) [inline test/test_utils.py]: Add the new functions here

**Comment by yzainee** [comment]
@animuk Please update your test cases as the test cov %age has gone down

File: src/utils.py (hunk 1 above)
Comment (animuk) [reply src/utils.py]: @yzainee this is a new f string structure in python. not a typo. you can also start using this.

File: src/utils.py (hunk 2 above)
Comment (animuk) [reply src/utils.py]: @yzainee logically the sync_all_non_cve_version function is used for some other purpose. As design wise, we should not merge the cve source code in the sync_all_non_cve_version() method. this is my take on that.

**Comment by animuk** [comment]
@yzainee : test coverage has gone up. Please check above
Merging #307 into master will increase coverage by 0.39%.

File: test/test_utils.py (hunk 4 above)
Comment (animuk) [reply test/test_utils.py]: added.

File: src/utils.py (hunk 3 above)
Comment (animuk) [reply src/utils.py]: added a new if condition.

File: src/utils.py (hunk 1 above)
Comment (yzainee) [reply src/utils.py]: ok.. wasnt aware of this :)

File: src/utils.py (hunk 2 above)
Comment (yzainee) [reply src/utils.py]: Change the function name [sync_all_non_cve_version] in that case. Make it generic.

**Comment by yzainee** [comment]
@animuk 

src/rest_api.py | 91.12% <100%> (-1.01%)
src/data_importer.py | 90.33% <0%> (-0.49%)

Overall coverage % is one thing. We also need to check the coverage % of each file as well.
Probably would let @tisnik comment on that. Because as per guidelines that he had put, thats what was decided.

**Comment by yzainee** [comment]
@animuk Also, please add in comments the total time of execution for this process ( on dev cluster & staging ). For any 1 time or periodical tasks, we need to know the time it takes to perform a particular activity.

File: src/utils.py (hunk 2 above)
Comment (animuk) [reply src/utils.py]: @yzainee Let my implementation separate. nothing wrong in that.

**Comment by tisnik** [comment]
OK from my side @animuk

**Comment by animuk** [comment]
Total Time of Execution of the process:  ~ 25 seconds
Logger output - 

INFO:src.utils:calculating execution time of the job:
  | INFO:src.utils:--- 25.052345514297485 seconds ---

cc @yzainee @SriKrishnaPaparaju

**Comment by animuk** [comment]
@yzainee please +1 to the PR if you feel that.

File: src/utils.py (hunk 2 above)
Comment (yzainee) [reply src/utils.py]: Repetitive code reduces the maintainability index. Thats the only reason i was asking to reuse the existing function.
current:
src/utils.py - A (37.89)
src/rest_api.py - A (40.00)

new(with the changes):
src/rest_api.py - A (39.21)
src/utils.py - A (35.93)

Im fine if @samuzzal-choudhury @tisnik are ok with it.

**Comment by SriKrishnaPaparaju** [comment]
@animuk what is the purpose of this PR ? Description is left blank

**Comment by animuk** [comment]
@miteshvp This is going to be invoked from Gemini. Please refer this - https://github.com/fabric8-analytics/fabric8-gemini-server/blob/master/src/rest_api.py#L58. Shortly will raise a PR for this

**Comment by animuk** [comment]
@SriKrishnaPaparaju Updated the description.

**Comment by miteshvp** [comment]
@animuk - as discussed, this changes has to be absolutely taken out once the job is done.

**Pull request merged**